services/supervisor/src/lib.rs

```diff
@@ -140,13 +199,12 @@ impl Service for Supervisor {
                 // Remove pending config proposal for which deadline was exceeded.
                 log::trace!("Removed outdated config proposal");
                 schema.pending_proposal().remove();
-            } else {
+            } else if entry.config_propose.actual_from == height.next() {
```

Comment:
Surprisingly previous implementation did apply configs not on the `actual_from` height, but at the next height instead. I've added a test to detect such a behaviour.